docs/plugin_author_guide/compiler_plugins.rst

```diff
@@ -0,0 +1,237 @@
+.. _compiler_plugins:
+
+Compiler Plugins
+================
+
+Most concrete algorithms in Metagraph are self-contained "black box
+implementations" that are run independently by the dispatcher or Dask
```

Comment:
The other docs in the Plugin Author Guide include "Prerequisite Reading" sections. I think this is the first mention of Dask in the Plugin Author Guide, so it might be good to add a "Prerequisite Reading" section with a link to the "Usage with Dask" doc.